### ISSUE TITLE
WPS exit with code 139 randomly

### ISSUE THREAD

**Issue by nboisteault** [3liz/py-qgis-wps#6]
Randomly when executing `make run` in tests directory i have those logs above.
Then in LWC I don't have algorithms displayed.

```
wps_1     | /usr/lib/python3/dist-packages/pyparsing.py:943: DeprecationWarning:
wps_1     | 
wps_1     | Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
wps_1     | 
wps_1     | /usr/lib/python3/dist-packages/pyparsing.py:3245: DeprecationWarning:
wps_1     | 
wps_1     | Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
wps_1     | 
wps_1     | /usr/share/qgis/python/plugins/processing/gui/wrappers.py:1757: DeprecationWarning:
wps_1     | 
wps_1     | invalid escape sequence \A
wps_1     | 
wps_1     | /usr/share/qgis/python/plugins/processing/gui/wrappers.py:1772: DeprecationWarning:
wps_1     | 
wps_1     | invalid escape sequence \A
wps_1     | 
wps_1     | 2020-06-03 07:53:47,147	[1]	ERROR	[qgis:1] Processing: Problem with SAGA installation: SAGA was not found or is not correctly installed
wps_1     | /usr/lib/python3/dist-packages/pyparsing.py:943: DeprecationWarning:
wps_1     | 
wps_1     | Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
wps_1     | 
wps_1     | /usr/lib/python3/dist-packages/pyparsing.py:3245: DeprecationWarning:
wps_1     | 
wps_1     | Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
wps_1     | 
wps_1     | /usr/share/qgis/python/plugins/processing/gui/wrappers.py:1757: DeprecationWarning:
wps_1     | 
wps_1     | invalid escape sequence \A
wps_1     | 
wps_1     | /usr/share/qgis/python/plugins/processing/gui/wrappers.py:1772: DeprecationWarning:
wps_1     | 
wps_1     | invalid escape sequence \A
wps_1     | 
wps_1     | 2020-06-03 07:53:47,147	[67]	ERROR	[qgis:67] Processing: Problem with SAGA installation: SAGA was not found or is not correctly installed
wps_1     | Traceback (most recent call last):
wps_1     |   File "/usr/share/qgis/python/plugins/processing/script/ScriptAlgorithmProvider.py", line 66, in load
wps_1     |     ScriptUtils.defaultScriptsFolder(),
wps_1     |   File "/usr/share/qgis/python/plugins/processing/script/ScriptUtils.py", line 51, in defaultScriptsFolder
wps_1     |     mkdir(folder)
wps_1     |   File "/usr/share/qgis/python/plugins/processing/tools/system.py", line 109, in mkdir
wps_1     |     os.mkdir(newdir)
wps_1     | FileExistsError: [Errno 17] File exists: '/home/qgis/.local/share/QGIS/QGIS3/profiles/default/processing/scripts'
wps_1     | 2020-06-03 07:53:47,177	[67]	INFO	[qgis:67] QGis processing initialized
wps_1     | 2020-06-03 07:53:47,178	[67]	DEBUG	= Registering provider 'script'
wps_1     | 2020-06-03 07:53:47,179	[67]	DEBUG	= Registering provider 'model'
wps_1     | 2020-06-03 07:53:47,185	[67]	INFO	Loaded plugin 'algorithms'
wps_1     | 2020-06-03 07:53:47,189	[67]	DEBUG	Starting ZMQ worker loop
tests_wps_1 exited with code 139

```

**Comment by dmarteau** [comment]
I confirm this is related with a race concurrency at startup: looks like the volume is not fully mounted at the time the check is performed. 

Appears only at startup, I let the issue open but I will do nothing atm since this is not a blocker.  
 
As a matter of facts, it appears only from time to time when running locally with docker-compose: The problem never occured on our Swarm managed infrastructure.

**Comment by dmarteau** [comment]
Closing this because there is no more activity on this issue.

**Comment by dmarteau** [comment]
Note: It is confirmed that the issue is related to a race condition in the  `os.mkdir(newdir)` method when several processes are initialized:
 
* A PR has been subitted to fix the problem: https://github.com/qgis/QGIS/pull/43006.   
* The base docker image has been fixed to include the patch until the 3.20 lands: https://github.com/3liz/docker-qgis-platform/commit/53a47924a0c5b20bac02fe3888269ab610df72d2
